pkg/test/options.go

```diff
@@ -0,0 +1,73 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package test
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/imdario/mergo"
+	"github.com/samber/lo"
+
+	"github.com/aws/karpenter-core/pkg/operator/options"
+)
+
+type OptionsFields struct {
+	// Vendor Neutral
+	ServiceName          *string
+	DisableWebhook       *bool
+	WebhookPort          *int
+	MetricsPort          *int
+	WebhookMetricsPort   *int
+	HealthProbePort      *int
+	KubeClientQPS        *int
+	KubeClientBurst      *int
+	EnableProfiling      *bool
+	EnableLeaderElection *bool
+	MemoryLimit          *int64
+	LogLevel             *string
+	BatchMaxDuration     *time.Duration
+	BatchIdleDuration    *time.Duration
+	FeatureGates         *options.FeatureGates
```

Comment:
I think we may need to create `test.FeatureGates` here that contains fields that are all pointers. I don't think we can determine if something is set like this since this will override the full featureGates struct even if there are multiple fields there and one of them wasn't set.

Reply:
That's true, I didn't really think about it since there's currently only a single field.